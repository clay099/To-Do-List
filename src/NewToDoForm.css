.NewTaskForm input {
	margin: 5px;
}
