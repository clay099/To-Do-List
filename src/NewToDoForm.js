import React, { useState } from "react";
import "./NewToDoForm.css";

const NewToDoForm = ({ addToDo }) => {
	const INITIALSTATE = { task: "" };
	const [formData, setFormData] = useState(INITIALSTATE);

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({ ...formData, [name]: value }));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		addToDo({ ...formData });
		setFormData(INITIALSTATE);
	};

	return (
		<form onSubmit={handleSubmit} className="NewTaskForm">
			<label htmlFor="task">New Task</label>
			<input
				id="task"
				name="task"
				placeholder="New Task"
				type="text"
				value={formData.task}
				onChange={handleChange}
			/>
			<button className="NewTaskForm-btn">Submit</button>
		</form>
	);
};

export default NewToDoForm;
