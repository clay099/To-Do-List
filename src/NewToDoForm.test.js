import React from "react";
import { render } from "@testing-library/react";
import NewToDoForm from "./NewToDoForm";

//smoke test
it("should render NewBoxForm", () => {
	render(<NewToDoForm />);
});

//snapshot test
it("should match snapshot", () => {
	const { asFragment } = render(<NewToDoForm />);
	expect(asFragment()).toMatchSnapshot();
});
