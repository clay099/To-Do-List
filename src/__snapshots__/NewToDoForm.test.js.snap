// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTaskForm"
  >
    <label
      for="task"
    >
      New Task
    </label>
    <input
      id="task"
      name="task"
      placeholder="New Task"
      type="text"
      value=""
    />
    <button
      class="NewTaskForm-btn"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
